src/Core/Validation/ValidationException.php

```diff
@@ -71,4 +95,24 @@ public function getResult()
     {
         return $this->result;
     }
+
+    /**
+     * Whether to show additional information in the error message depending on the context
+     *
+     * We do not want this to show this in a context where it's easy to know the record and value that
+     * triggered the error e.g. form submission, API calls, etc
+     */
+    private function doShowAdditionalInfo(): bool
+    {
+        if (Director::is_cli()) {
+            return true;
+        }
+        $currentController = Controller::curr();
+        foreach ($this->config()->get('show_additional_info_non_cli_controllers') as $controller) {
```

Comment:
```suggestion
        foreach (static::config()->get('show_additional_info_non_cli_controllers') as $controller) {
```
Sorry, should have noticed this last round. `config()` is a static method - the calls to `$this->config()` that already exist are incorrect and should eventually be cleaned up - but certainly new code shouldn't call the static method non-statically.

Reply:
Updated